tests/src/Interop/PInvoke/Array/MarshalArrayAsParam/LPArrayNative/MarshalArrayLPArrayNative.cpp

```diff
@@ -5,6 +5,12 @@
 #include <xplatform.h>
 #include "MarshalArray.h"
 
+#if defined(_MSC_VER) || defined(__llvm__)
+#define FUNCTIONNAME __FUNCSIG__
+#else
+#define FUNCTIONNAME __func__
+#endif //_MSC_VER
```

Comment:
Based on https://gcc.gnu.org/onlinedocs/gcc/Function-Names.html, perhaps it would be better to prefer `__PRETTY_FUNCTION__` to match the output of MSVC; as `__func__` only returns the function name, not the whole signature:

```c
#ifdef __GNUC__
#define FUNCTIONNAME __PRETTY_FUNCTION__
#else // __GNUC__
#define FUNCTIONNAME  __FUNCSIG__
#endif // __GNUC__
```

demo: [clang](https://rextester.com/IOEJCR22400), [gcc](https://rextester.com/ZJSP73839), [msvc](https://rextester.com/AJX87709)

Reply:
done